src/util/uuid/mod.rs

```diff
@@ -0,0 +1,110 @@
+use std::char;
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+pub fn uuid() -> String {
+    let mut numbers = [0u8; 36];
+    make_random_numbers(&mut numbers);
+    uuid_from_numbers(&numbers)
+}
+
+#[cfg(target_arch = "wasm32")]
+fn make_random_numbers(numbers: &mut [u8]) {
+    web_sys::window()
+        .expect("window is not available")
+        .crypto()
+        .expect("window.crypto is not available")
+        .get_random_values_with_u8_array(numbers)
+        .expect("window.crypto.getRandomValues not available");
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+fn make_random_numbers(numbers: &mut [u8]) {
+    use rand::Rng;
+    let mut rng = rand::thread_rng();
+    for v in numbers.iter_mut() {
+        *v = rng.gen();
+    }
+}
+
+enum UuidElements {
+    Random09AF,
+    Random89AB,
+    Hyphen,
+    Version,
+}
+
+const UUID_V4_FORMAT: [UuidElements; 36] = [
+    UuidElements::Random09AF,
+    UuidElements::Random09AF,
+    UuidElements::Random09AF,
+    UuidElements::Random09AF,
+    UuidElements::Random09AF,
+    UuidElements::Random09AF,
+    UuidElements::Random09AF,
+    UuidElements::Random09AF,
+    UuidElements::Hyphen,
+    UuidElements::Random09AF,
+    UuidElements::Random09AF,
+    UuidElements::Random09AF,
+    UuidElements::Random09AF,
+    UuidElements::Hyphen,
+    UuidElements::Version,
+    UuidElements::Random09AF,
+    UuidElements::Random09AF,
+    UuidElements::Random09AF,
+    UuidElements::Hyphen,
+    UuidElements::Random89AB,
+    UuidElements::Random09AF,
+    UuidElements::Random09AF,
+    UuidElements::Random09AF,
+    UuidElements::Hyphen,
+    UuidElements::Random09AF,
+    UuidElements::Random09AF,
+    UuidElements::Random09AF,
+    UuidElements::Random09AF,
+    UuidElements::Random09AF,
+    UuidElements::Random09AF,
+    UuidElements::Random09AF,
+    UuidElements::Random09AF,
+    UuidElements::Random09AF,
+    UuidElements::Random09AF,
+    UuidElements::Random09AF,
+    UuidElements::Random09AF,
+];
+
+const ERROR_MAKE_CHAR: &str = "Error in making char";
+
+pub fn uuid_from_numbers(random_numbers: &[u8; 36]) -> String {
+    UUID_V4_FORMAT
+        .iter()
+        .enumerate()
+        .map(|(i, kind)| match kind {
+            UuidElements::Random09AF => {
+                char::from_digit((random_numbers[i] & 0b1111) as u32, 16).expect(ERROR_MAKE_CHAR)
+            }
+            UuidElements::Random89AB => {
+                char::from_digit((random_numbers[i] & 0b11) as u32 + 8, 16).expect(ERROR_MAKE_CHAR)
+            }
+            UuidElements::Version => '4',
```

Comment:
curious where this version comes from?

Reply:
It's part of the spec. @arv being @arv, he implemented the actual uuid spec rather than a random number which is all we needed :).

Reply:
In particular v4 of the uuid spec. Yes there have been 4 (actually 5) versions of a spec for a large unique number.

Reply:
This was taken from somewhere... I think the original PR had a link to it.

If we only need a random number we cam simplify.

Reply:
How big does this really need to be @phritz . IIRC clientID is scoped within a particular account. Two random 64b numbers should more than do it, right? Should we go ahead and simplify this?

Reply:
https://github.com/rocicorp/repc/issues/183